src/PostgREST/Statements.hs

```diff
@@ -72,6 +72,11 @@ createWriteStatement selectQuery mutateQuery wantSingle isInsert asCsv rep pKeys
     | wantSingle = asJsonSingleF
     | otherwise = asJsonF
 
+  selectF
+    -- prevent using any of the column names in ?select= when no response is returned from the CTE
+    | rep `elem` [None, HeadersOnly] = "SELECT * FROM " <> sourceCTEName
```

Comment:
Hmm.. here I think a `SELECT null` would suffice for preventing the error. With this you wouldn't need the change made to the `QueryBuilder` I think.

Reply:
I think for the HeadersOnly case you still need the columns at this point to put together the location string or am I missing something right now?

And that makes me wonder: Is it actually guaranteed that the PK columns are always in the `RETURNING` part?

Reply:
Like when I do a `POST return=rep select=non_pk_only`.

I remember the foreign keys for the embedded columns get added to the `RETURNING`, but I am not sure about the PKs.

Reply:
> Is it actually guaranteed that the PK columns are always in the RETURNING part?

No, but maybe that should be a different fix, check https://github.com/PostgREST/postgrest/issues/1461.

Reply:
added some test cases for that at #1475 
it's not throwing, but just not returning a location header at all it seems. So yes, this is a problem.